enterprise_access/apps/content_assignments/models.py

```diff
@@ -675,17 +675,30 @@ def add_successful_acknowledged_expired_action(self):
             completed_at=timezone.now(),
         )
 
+    @staticmethod
+    def _unique_retired_email():
+        """
+        Helper to return a templated email address
+        that's relatively uniqueified with the addition of a random, 8-byte,
+        hex string.
+        """
+        nonce = urandom(8).hex()
+        return RETIRED_EMAIL_ADDRESS_FORMAT.format(nonce)
+
     def clear_pii(self):
         """
-        Removes PII field values from this assignment.
+        Removes PII field values from this assignment by setting
+        the ``learner_email`` field to a templated email address
+        that's relatively uniqueified with the addition of a random, 8-byte,
+        hex string.
         """
-        self.learner_email = RETIRED_EMAIL_ADDRESS
+        self.learner_email = self._unique_retired_email()
 
     def clear_historical_pii(self):
         """
         Removes PII values from this assignment's historical records.
         """
-        self.history.update(learner_email=RETIRED_EMAIL_ADDRESS)  # pylint: disable=no-member
+        self.history.update(learner_email=self._unique_retired_email())  # pylint: disable=no-member
```

Comment:
[sanity check, question] Are there any concerns around the history tables' redacted email addresses likely not matching the non-history tables' redacted email for the same assignment due to the random hex string?

Reply:
Good suggestion, I just modified `clear_pii()` to retire the main record and the historical records with the same value.